### DISCUSSION TITLE
Support RFC 7986 properties from the .ics file

### DISCUSSION THREAD

**Discussion by Yenya** [bitfireAT/icsx5#187] [Ideas]
RFC 7986 (https://www.rfc-editor.org/rfc/rfc7986) allows most of the global attributes of the calendar to be defined inside the .ics file. It would be nice if icsx5 can read the ics file first, and preset the attributes provided there into the settings dialog:

- SOURCE or URL - can be used to set the real URL of the calendar to get further updates from, when for example importing the .ics from the local copy received by other means (e.g. bluetooth share)
- COLOR
- NAME or DESCRIPTION - the calendar title
- REFRESH-INTERVAL

Thanks for considering this.

**Comment by cbix** [comment]
`NAME` and `COLOR` are already recognized and set correctly, never tested the others.

**Comment by ArnyminerZ** [comment]
We are not sure about this. As @cbix has pointed out, `NAME` and `COLOR` are supported. I think that `SOURCE` and `URL` might be useful, but they require quite a big infrastructure change, because the main use (pointing to Internet locations instead of local files), implicate two different sources of the data, which may break a lot of things.

So, I don't think we will work on this, at least as of right now. If enough people ask for it, we of course can reconsider our decision. Hope you understand.

**Reply by cbix** [reply]
> implicate two different sources of the data

I understand the spec like this:
1. ics is parsed and updated normally
2. if ics contains `SOURCE` or `URL`, _after_ the sync the feed settings are updated to point to the new URL
3. the new URL is only fetched on the next sync
